types/coin.go

```diff
@@ -0,0 +1,19 @@
+package types
+
+import (
+	basev1beta1 "github.com/cosmos/cosmos-sdk/api/cosmos/base/v1beta1"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func ProtoCoinsToCoins(coins []*basev1beta1.Coin) (sdk.Coins, bool) {
+	result := make([]sdk.Coin, 0, len(coins))
+	for _, coin := range coins {
+		amount, ok := sdk.NewIntFromString(coin.Amount)
+		if !ok {
+			return nil, ok
+		}
+		result = append(result, sdk.NewCoin(coin.Denom, amount))
+	}
+
+	return result, true
+}
```

Comment:
can we add a godoc here? also might be worth adding to the doc that this function _could_ panic. ((`NewCoin` panics on invalid denoms))